core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -100,6 +100,12 @@ object ParserGen {
       Gen.const(GenT(Parser.anyChar))
     )
 
+  val stringIn1: Gen[GenT[Parser1]] =
+    Arbitrary.arbitrary[List[String]].map { cs =>
+      if (cs.size < 2 || cs.exists(_.isEmpty)) GenT(Parser.fail: Parser1[Unit])
```

Comment:
I don't think you need the cs.size check anymore since you special cased the constructor method.

Reply:
Not this case, this makes sure that we don't have the empty string in alternatives as this is a `Parser1`, it cannot accept the empty string.

Reply:
but you also do a size check in stringIn1, so it is checking if size is < 2 (if 0 it returns fail, if 1 it returns stringIn).

Also, since these could in principle be two duplicate items, I think it is better to remove the size check here and just let stringIn1 do its work.

Reply:
Oh sorry I misread, of course this test is now not used anymore. I will remove it.